integration-cli/docker_cli_wait_test.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"os/exec"
+	"testing"
+)
+
+// non-blocking wait with 0 exit code
+func TestWaitNonBlockedExitZero(t *testing.T) {
+	defer deleteAllContainers()
+
+	runCmd := exec.Command(dockerBinary, "run", "-d", "busybox", "sh", "-c", "true")
+	out, _, err := runCommandWithOutput(runCmd)
+	if err != nil {
+		t.Fatal(out, err)
+	}
+	containerID := stripTrailingCharacters(out)
+
+	status := "true"
+	for status != "false" {
+		runCmd = exec.Command(dockerBinary, "inspect", "--format='{{.State.Running}}'", containerID)
+		status, _, err = runCommandWithOutput(runCmd)
+		if err != nil {
+			t.Fatal(status, err)
+		}
```

Comment:
you may need to strip all spaces (newlines, etc..) from 'status' - the CI seems to hang on this.